cmd/juju/charmhub/info.go

```diff
@@ -107,17 +115,20 @@ func (c *infoCommand) Init(args []string) error {
 		return errors.Errorf("unexpected unicode flag value %q, expected <auto|never|always>", c.unicode)
 	}
 
-	switch c.arch {
-	case "all", "amd64", "arm64", "ppc64", "s390":
-	case "":
-		c.arch = "all"
-	default:
-		return errors.Errorf("unexpected architecture flag value %q, expected <all|amd64|arm64|ppc64|s390>", c.arch)
+	// If the architecture is empty, ensure we normalize it to all to prevent
+	// complicated comparison checking.
+	if c.arch == "" {
+		c.arch = ArchAll
```

Comment:
See my comments below about ArchAll. Also, this should probably read  `c.arch = corecharm.ArchAll`